projects/plugins/boost/app/assets/src/js/modules/image-size-analysis/recommendations/row-types/ImageSizeRow.svelte

```diff
@@ -24,6 +28,22 @@
 
 	const sizeDifference = ( potentialSavings / currentSize ) * 100;
 	const pillColor = sizeDifference <= 30 ? '#f5e5b3' : '#facfd2';
+
+	async function fixImageSize() {
+		const data = {
+			image_url: details.image.url,
+			image_width: details.image.dimensions.expected.width.toString(),
+			image_height: details.image.dimensions.expected.height.toString(),
+			post_id: post_id.toString(),
+			nonce: Jetpack_Boost.fixImageNonce,
+		};
+		return await api.post( '/image-size-analysis/fix', data );
+	}
```

Comment:
I think we'll try to deprecate `api.post()` during the refactor in favor of using DataSync - perhaps we should use DataSync here so that we don't have to refactor this later?

Reply:
We're going with api.post() just to get this merged, and then refactor it along with everything else afterwards.